### DISCUSSION TITLE
Examples of how to generate refresh token?

### DISCUSSION THREAD

**Discussion by danni-popova** [lestrrat-go/jwx#383] [Q&A] [answered]
Hi,

I've found all of the provided examples super useful and have manage to integrate this into my own system. 

I am looking for some examples of the token refresh flow though and was wondering if you have any pointers?

**Comment by lestrrat** [comment, marked as answer]
Does this work for you? https://github.com/lestrrat-go/jwx/blob/main/examples/jwk_example_test.go#L188

**Reply by danni-popova** [reply]
Maybe I've understood this incorrectly, but aren't the JWKs the public keyset that we retrieve in order to validate an access token? [Reference](https://datatracker.ietf.org/doc/html/rfc7517)


And what I am trying to understand is how to genereate a  [refresh](https://auth0.com/docs/tokens/refresh-token/current) token alongside the access token.

**Reply by lestrrat** [reply]
Oh, I thought you wanted to refresh the public JWKs that you obtained.

A refresh token is issued by the service provider, and is not in the scope of this library.  I mean, it's usually just a string key of sorts that you send to the provider. Think of it as sort of a cache key. You authenticate once, and you get a refresh token that you can keep reusing to get a new access token until the refresh token is invalidated.
